lib/src/room.dart

```diff
@@ -1187,22 +1187,18 @@ class Room {
   /// Set the power level of the user with the [userID] to the value [power].
   /// Returns the event ID of the new state event. If there is no known
   /// power level event, there might something broken and this returns null.
-  Future<String> setPower(String userID, int power) async {
-    final powerMap = Map<String, Object?>.from(
-      getState(EventTypes.RoomPowerLevels)?.content ?? {},
-    );
-
-    final usersPowerMap = powerMap['users'] is Map<String, Object?>
-        ? powerMap['users'] as Map<String, Object?>
-        : (powerMap['users'] = <String, Object?>{});
+  Future<String> setPower(String userId, int power) async {
+    final powerLevelMapCopy =
+        getState(EventTypes.RoomPowerLevels)?.content.copy() ?? {};
 
-    usersPowerMap[userID] = power;
+    powerLevelMapCopy['users'] ??= <String, Object?>{};
+    powerLevelMapCopy.tryGetMap<String, Object?>('users')?[userId] = power;
```

Comment:
This operates on a copy of `powerLevelMapCopy`, so this actually never modifies the power levels of that user! That's probably also why the tests hang, but for me at least the command test with the op command explicitly fails. Maybe we can add a proper test for this?

Reply:
this is also a copy? 😳 oopsie...